IPython/external/qt_loaders.py

```diff
@@ -110,24 +110,29 @@ def has_binding(api):
                    QT_API_PYQT_DEFAULT: 'PyQt4'}
     module_name = module_name[api]
 
-    import imp
+
+    if sys.version_info >= (3, 4):
+        from importlib.utils import find_spec as find_module_or_spec
+    else :
+        from imp import find_module as find_module_or_spec
+
     try:
         #importing top level PyQt4/PySide module is ok...
         mod = __import__(module_name)
         #...importing submodules is not
-        imp.find_module('QtCore', mod.__path__)
-        imp.find_module('QtGui', mod.__path__)
-        imp.find_module('QtSvg', mod.__path__)
+        find_module_or_spec('QtCore', mod.__path__)
+        find_module_or_spec('QtGui', mod.__path__)
+        find_module_or_spec('QtSvg', mod.__path__)
         if api == QT_API_PYQT5:
             # QT5 requires QtWidgets too
-            imp.find_module('QtWidgets', mod.__path__)
+            find_module_or_spec('QtWidgets', mod.__path__)
 
         #we can also safely check PySide version
         if api == QT_API_PYSIDE:
             return check_version(mod.__version__, '1.0.3')
         else:
             return True
-    except ImportError:
+    except ImportError, ValueError:
```

Comment:
You need parentheses around these, otherwise it's invalid syntax on Python 3, and not doing what you expect on Python 2.